src/main/scala/firrtl/Mappers.scala

```diff
@@ -79,7 +79,10 @@ object Mappers {
           case s: DefWire => DefWire(s.info,f(s.name),s.tpe)
           case s: DefPoison => DefPoison(s.info,f(s.name),s.tpe)
           case s: DefRegister => DefRegister(s.info,f(s.name), s.tpe, s.clock, s.reset, s.init)
-          case s: DefMemory => DefMemory(s.info,f(s.name), s.data_type, s.depth, s.write_latency, s.read_latency, s.readers, s.writers, s.readwriters)
+          case s: DefMemory =>
+            DefMemory(s.info,f(s.name), s.data_type, s.depth, s.write_latency,
```

Comment:
This is tricky - I'm not sure you should map onto the names of the ports, because generally, mapping String=>String on a statement should only change the name of the statement, but not to any types in the design (the type of a reference to this memory would change). I suppose another way to do this would be to do something in forType's map(Stmt):Stmt, where the data type of the DefMemory is created from its readers/writers/readwriters and data_type, passed into the mapper function, and then map any changes back.